homeassistant/components/airtouch5/climate.py

```diff
@@ -0,0 +1,388 @@
+    )
+
+    def __init__(
+        self, client: Airtouch5SimpleClient, name: ZoneName, ac: AcAbility
+    ) -> None:
+        """Initialise the Climate Entity."""
+        super().__init__(client)
+        self._name = name
+
+        self._attr_unique_id = f"zone_{name.zone_number}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, f"zone_{name.zone_number}")},
+            name=name.zone_name,
+            manufacturer="Polyaire",
+            model="AirTouch 5",
+        )
+        # We can have different setpoints for heat and cool, we expose the lowest low and highest high
+        self._attr_min_temp = min(ac.min_cool_set_point, ac.min_heat_set_point)
+        self._attr_max_temp = max(ac.max_cool_set_point, ac.max_heat_set_point)
+
+    @callback
+    def _async_update_attrs(self, data: dict[int, ZoneStatusZone]) -> None:
+        if self._name.zone_number not in data:
+            return
+        status = data[self._name.zone_number]
+        self._attr_current_temperature = status.temperature
+        self._attr_target_temperature = status.set_point
+
+        if status.zone_power_state == ZonePowerState.OFF:
+            self._attr_hvac_mode = HVACMode.OFF
+            self._attr_preset_mode = PRESET_NONE
+        elif status.zone_power_state == ZonePowerState.ON:
+            self._attr_hvac_mode = HVACMode.FAN_ONLY
+            self._attr_preset_mode = PRESET_NONE
+        elif status.zone_power_state == ZonePowerState.TURBO:
+            self._attr_hvac_mode = HVACMode.FAN_ONLY
+            self._attr_preset_mode = PRESET_BOOST
+        else:
+            self._attr_hvac_mode = None
+
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Add data updated listener after this object has been initialized."""
+        await super().async_added_to_hass()
+        self._client.zone_status_callbacks.append(self._async_update_attrs)
+        self._async_update_attrs(self._client.latest_zone_status)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Remove data updated listener after this object has been initialized."""
+        await super().async_will_remove_from_hass()
+        self._client.zone_status_callbacks.remove(self._async_update_attrs)
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new operation mode."""
+        power: ZoneSettingPower
+
+        if hvac_mode is HVACMode.OFF:
+            power = ZoneSettingPower.SET_TO_OFF
+        elif self._attr_preset_mode is PRESET_BOOST:
+            power = ZoneSettingPower.SET_TO_TURBO
+        else:
+            power = ZoneSettingPower.SET_TO_ON
+
+        zcz = ZoneControlZone(
+            self._name.zone_number,
+            ZoneSettingValue.KEEP_SETTING_VALUE,
+            power,
+            0,
+        )
+        packet = self._client.data_packet_factory.zone_control([zcz])
+        await self._client.send_packet(packet)
```

Comment:
Same here, please move it to a function to reduce duplicate code

Reply:
Same here

Reply:
Please adopt the code similar to my example above